controllers/component_build_controller.go

```diff
@@ -544,7 +544,7 @@ func ConfigureRepositoryForPaC(component appstudiov1alpha1.Component, config map
 	}
 }
 
-func GeneratePipelineRun(name, namespace, bundle, image string, onPull bool) []byte {
+func GeneratePipelineRun(component appstudiov1alpha1.Component, bundle string, onPull bool) []byte {
```

Comment:
Since more additional component properties are read from component.Spec, the `GeneratePipelineRun` function is refactored to accept a Component object in order to reduce the number of arguments. Is there any other function call to `GeneratePipelineRun` from outside build-service?

Reply:
looks good to me, there is no external dependency